internal/gatewayapi/route.go

```diff
@@ -1468,15 +1471,16 @@ func (t *Translator) processServiceDestinationSetting(
 		endpoints, addrType = getIREndpointsFromEndpointSlices(endpointSlices, servicePort.Name, servicePort.Protocol)
 	} else {
 		// Fall back to Service ClusterIP routing
-		ep := ir.NewDestEndpoint(service.Spec.ClusterIP, uint32(*backendRef.Port), false)
+		ep := ir.NewDestEndpoint(service.Spec.ClusterIP, uint32(*backendRef.Port), false, nil)
```

Comment:
we can optimize with new function named `ir. NewDestEndpointWithZone()`

Reply:
Currently only Kubernetes Services are supported but wanted to leave this open in case `Backend` or other backend types had a way to provide zone information in the future.

Reply:
it's not a blocker.